test/fixtures/runtime_dns.js

```diff
@@ -0,0 +1,27 @@
+const assert = require('assert');
+const dns = require('dns');
+const sinon = require('sinon');
+
+exports.handler = (event, context, callback) => {
+  const failure = new Error('simulated failure');
+  failure.code = dns.NOTFOUND;
+
+  const lookup = sinon.stub(dns.lookup, '_raw')
+    .callsArgWith(2, failure)
+    .withArgs('example.com', sinon.match.object, sinon.match.func)
+    .onFirstCall().callsArgWith(2, failure)
+    .onSecondCall().callsArgWith(2, failure)
+    .onThirdCall().callsArgWith(2, null, '127.0.0.1', 4);
+
+  dns.lookup('example.com', (error, hostname, family) => {
+    try {
+      assert.ifError(error);
+      assert.equal(hostname, '127.0.0.1');
+      assert.equal(family, 4);
+      sinon.assert.callCount(lookup, 3);
+      callback(null, 'success!');
```

Comment:
Minor point, but a pattern I've used for callbacks in try/catch blocks is:

```
try {
  something
} catch (e) {
  callback(e);
  return;
}
callback(null, success)
```

The current code can cause a double callback if the `callback(null, 'success!');` line throws an Error. That case becomes very hard to debug when it happens.

Reply:
Good point. I'll update this. Thanks.